### ISSUE TITLE
"-DHTTP_ONLY=ON" in third_party/curl/CMakeLists.txt

### ISSUE THREAD

**Issue by tlapik123** [mavlink/MAVSDK#2045]
Is this intentional?

Since if the curl is build without https support the https .xml definition files sent by the camera will error out.
When the support it switched on it download the .xml correctly.

But maybe there is some issue for this that I have missed that explained why it is this way? :)

**Comment by JonasVautherin** [comment]
I don't remember... probably it's just that OpenSSL is (was?) not a dependency if curl doesn't use HTTPS.

But then if you want to use HTTPS, you need a valid certificate on the drone. Do you have that?

**Comment by tlapik123** [comment]
> I don't remember... probably it's just that OpenSSL is (was?) not a dependency if curl doesn't use HTTPS.
Is there any reason against using openSSL in this case? Less space/code?

> But then if you want to use HTTPS, you need a valid certificate on the drone. Do you have that?
I'm developing a test application and running that on the PC with the camera connected (through ardupilot). So this is not an issue for me.

**Comment by JonasVautherin** [comment]
> Is there any reason against using openSSL in this case? Less space/code?

If it's not needed, it's always nice to have fewer dependencies. Also for our superbuild scheme, it's faster to build.

**Comment by JonasVautherin** [comment]
Yeah it is used for camera_definition.xml files served as http :+1:
